util/container/Dockerfile

```diff
@@ -4,174 +4,127 @@
-    mkdir -p riscv-llvm && \
-    echo "SNITCH_LLVM_VERSION=${SNITCH_LLVM_VERSION} LLVM_TAR=${LLVM_TAR} latest_tag=${latest_tag}" && \
-    wget -qO- https://github.com/pulp-platform/llvm-project/releases/download/${SNITCH_LLVM_VERSION}/${LLVM_TAR} | \
-    tar xvz --strip-components=1 -C riscv-llvm
-ENV LLVM_BINROOT "/tools/riscv-llvm/bin"
-
-# Install Verible
-RUN wget https://github.com/chipsalliance/verible/releases/download/v${VERIBLE_VERSION}/verible-v${VERIBLE_VERSION}-linux-static-x86_64.tar.gz && \
-    tar -x -f verible-v${VERIBLE_VERSION}-linux-static-x86_64.tar.gz --strip-components=1 -C . && \
-    rm -rf verible-v${VERIBLE_VERSION}-linux-static-x86_64.tar.gz
-ENV PATH "/tools/bin:${PATH}"
-
-# Install git>=2.18, required by Github checkout action to recurse submodules
-RUN apt-get update && apt-get install software-properties-common -y && \
-    add-apt-repository -y ppa:git-core/ppa && \
-    apt-get update && \
-    apt-get install git -y
-
-# Copy artifacts from stage 1.
-COPY --from=builder /tools/bender bin/
-COPY --from=builder /tools/spike-dasm bin/
-COPY --from=builder /root/.cargo/bin/banshee bin/
-COPY --from=builder /opt/python /opt/python
-COPY --from=builder /tools/verilator /tools/verilator/
-COPY --from=builder /tools/doxygen/bin/doxygen bin/
+# Copy all needed files to install the package
+COPY pyproject.toml .
+COPY --from=python_filter sw ./sw
+COPY --from=python_filter util ./util
 
 # Create and activate virtual environment
-ENV VIRTUAL_ENV "/root/.venvs/snitch_cluster"
-RUN /opt/python/bin/python3 -m venv ${VIRTUAL_ENV}
+ARG VIRTUAL_ENV="/root/.venvs/snitch_cluster"
+RUN python${PYTHON_VERSION} -m venv ${VIRTUAL_ENV}
+
+# Upgrade pip and install packages using the virtual environment's pip
+RUN ${VIRTUAL_ENV}/bin/pip install --upgrade pip && \
+    ${VIRTUAL_ENV}/bin/pip install .
+
+# Copy the tools from the builder stage
+COPY --from=builder /tools/bender /tools/bin/
+COPY --from=builder /root/.cargo/bin/banshee /tools/bin/
+COPY --from=builder /tools/doxygen/bin/doxygen /tools/bin/
+COPY --from=builder /tools/verible* /tools/bin/
+COPY --from=builder /tools/riscv-llvm /tools/riscv-llvm
+
+# Modify environment and path variables
+ENV VLT="verilator-wrap.sh"
+ENV LLVM_BINROOT="/tools/riscv-llvm/bin"
 ENV PATH="${VIRTUAL_ENV}/bin:${PATH}"
-# Install Python requirements
-COPY python-requirements.txt /tmp/python-requirements.txt
-COPY docs/requirements.txt /tmp/docs/requirements.txt
-COPY sw/dnn/requirements.txt /tmp/sw/dnn/requirements.txt
-RUN pip install -r /tmp/python-requirements.txt
-
-# Add Verilator to PATH
-ENV PATH "/tools/verilator/bin:${PATH}"
-ENV VLT_ROOT "/tools/verilator/share/verilator/"
-
-# Set locale to UTF-8, required because Python 3.6 defaults on ASCII encoding.
-# See https://click.palletsprojects.com/en/8.1.x/unicode-support/
-ENV LC_ALL "C.UTF-8"
-ENV LANG   "C.UTF-8"
+ENV PATH="/tools/bin:${PATH}"
+ENV PATH="/tools/riscv-llvm/bin:${PATH}"
```

Comment:
Is it necessary to add `LLVM_BINROOT` to the `PATH`? Everything should work even without, as we don't rely on this on IIS machines.

Reply:
I think I only added it for `clang-format`